tests/testSpherePoint.py

```diff
@@ -0,0 +1,751 @@
+            ((90.0, 0.0), afwGeom.Point3D(0.0, 1.0, 0.0)),
+            ((0.0, 90.0), afwGeom.Point3D(0.0, 0.0, 1.0)),
+        ]
+
+        for lonLat, vector in pointList:
+            # Convert to Point3D.
+            point = SpherePoint(lonLat[0]*degrees, lonLat[1]*degrees)
+            newVector = point.getVector()
+            self.assertIsInstance(newVector, afwGeom.Point3D)
+            for oldElement, newElement in zip(vector, newVector):
+                self.assertAlmostEqual(oldElement, newElement)
+
+            # Convert back to spherical.
+            newLon, newLat = SpherePoint(newVector)
+            self.assertAlmostEqual(newLon.asDegrees(), lonLat[0])
+            self.assertAlmostEqual(newLat.asDegrees(), lonLat[1])
+
+        # Try some un-normalized ones, too.
+        pointList = [
+            ((0.0, 0.0), afwGeom.Point3D(1.3, 0.0, 0.0)),
+            ((90.0, 0.0), afwGeom.Point3D(0.0, 1.2, 0.0)),
+            ((0.0, 90.0), afwGeom.Point3D(0.0, 0.0, 2.3)),
+            ((0.0, 0.0), afwGeom.Point3D(0.5, 0.0, 0.0)),
+            ((90.0, 0.0), afwGeom.Point3D(0.0, 0.7, 0.0)),
+            ((0.0, 90.0), afwGeom.Point3D(0.0, 0.0, 0.9)),
+        ]
+
+        for lonLat, vector in pointList:
+            # Only convert from vector to spherical.
+            point = SpherePoint(vector)
+            newLon, newLat = point
+            self.assertAlmostEqual(lonLat[0], newLon.asDegrees())
+            self.assertAlmostEqual(lonLat[1], newLat.asDegrees())
+            self.assertAlmostEqual(1.0, point.getVector().distanceSquared(afwGeom.Point3D(0.0, 0.0, 0.0)))
+
+        # Ill-defined points should be all NaN after normalization
+        pointList = [
+            afwGeom.Point3D(0.5, nan, 0.0),
+            afwGeom.Point3D(0.5, 0.0, -inf),
+        ]
+        for vector in pointList:
+            point = SpherePoint(vector)
+            for element in point.getVector():
+                self.assertTrue(math.isnan(element))
+
+    def testTicket1761(self):
+        """Regression test for Ticket 1761.
+
+        Checks for math errors caused by unnormalized vectors.
+        """
+        refPoint = SpherePoint(afwGeom.Point3D(0, 1, 0))
+
+        point1 = SpherePoint(afwGeom.Point3D(0.1, 0.1, 0.1))
+        point2 = SpherePoint(afwGeom.Point3D(0.6, 0.6, 0.6))
+        sep1 = refPoint.separation(point1)
+        sep2 = refPoint.separation(point2)
+        sepTrue = 54.735610317245339*degrees
+
+        self.assertAnglesNearlyEqual(sepTrue, sep1)
+        self.assertAnglesNearlyEqual(sepTrue, sep2)
+
+    def testAtPoleValue(self):
+        """Test if atPole() returns the expected value.
+        """
+        poleList = [
+            SpherePoint(0.0*degrees, -90.0*degrees),
+            SpherePoint(42.0*degrees, 324000.0*afwGeom.arcseconds),
+            SpherePoint(-386.0*degrees, afwGeom.HALFPI*radians),
+            SpherePoint(afwGeom.Point3D(0.0, 0.0, 1.0)),
+            SpherePoint(afwGeom.Point3D(0.0, 0.0, -1.0)),
```

Comment:
As mentioned earlier, please test both poles for all units. I don't care about longitude, but I care a lot about latitude.

Also, test just inside of the atPole limit for both poles, all units. Consider scaling epsilon by the max value (e.g. epsilon/90 for degrees, etc.) for unit-independent resolution.